src/me/drton/jmavsim/Simulator.java

```diff
@@ -732,6 +760,36 @@ public static void main(String[] args)
                     System.err.println("-qgc needs an argument: " + QGC_STRING);
                     return;
                 }
+            } else if (arg.equals("-sdk")) {
+                COMMUNICATE_WITH_SDK = true;
+                if (i == args.length) {
+                    // only arg is -sdk, so use default values.
+                    break;
+                }
+                if (i < args.length) {
+                    String nextArg = args[i++];
+                    if (nextArg.startsWith("-")) {
+                        // only turning on udp, but want to use default ports
+                        i--;
+                        continue;
+                    }
+                    try {
+                        // try to parse passed-in ports.
+                        String[] list = nextArg.split(":");
+                        if (list.length != 2) {
+                            System.err.println("Expected: " + QGC_STRING + ", got: " + Arrays.toString(list));
```

Comment:
Should be `SDK_STRING`, I believe. But that's sooo similar to the QGC parsing that I would put that whole block in a function ;-)

Reply:
Thanks!